doc/adr/0009-use-crds-in-declarative-interface.md

```diff
@@ -0,0 +1,74 @@
+# 9. Use CRDs in declarative interface
+
+Date: 2024-02-26
+
+## Status
+
+Proposed
+
+## Context
+
+The motivation for this decision is a desire to provide a better, more
+intuitive interface between the controller and the user. Custom
+Resource Definitions are now the standard way of providing extended
+interfaces in Kubernetes. They are as good as any other format for
+podman based sites.
+
+In addition to allowing the user to configure skupper through more
+clearly defined types, CRDs also give us a familiar way to get
+information back from the controller to the user through the status
+sections.
+
+Only the controller image should be tied to using the CRDs
+directly. The config-sync image should not need to interact with
+them. That ensures that should other aprpoaches be necessary we
+minimise the amount of work required.
+
+## Decision
+
+Instead of creating ConfigMaps and Secrets with annotations users will
+create custom resources to configure Skupper.
+
+## Consequences
+
+Standard RBAC can be used to limit, at a coarse granularity, what
```

Comment:
On podman sites, if CRs are stored as regular files (which is my current inclination so far), those files must have been granted write permission to the current user id, as podman sites will need permissions to update their statuses.
Would it be fair to assume that an eventual filesystem directory can be owned by a certain unix group and then a similar limitation could be applied as well?

Reply:
A podman site is single tenant, so I'm not sure how important any such restrictions would be? The host may allow separate users, and I guess you may then want some control over what users can access from the host machine, but I think that would be through the regular OS controls perhaps?

Reply:
Yes, my assumption is that people will see it as a single tenant as well and rely on OS related permissions to control that, not expecting skupper to do so.